### ISSUE TITLE
No way to create WCAG-compliant alt text for lead images : 5

### ISSUE THREAD

**Issue by paregorios** [isawnyu/isaw.web#296]
Steps to recreate:

 - Create a new page
 - Specify a lead image
 - Fill out the lead image caption field
 - Save

Note the resulting markup:

```html
<img src="https://isaw.nyu.edu/members/te20-40nyu.edu/alt_test/lead_image_only/leadImage_mini" alt="Lead Image Only" title="Lead Image Only" class="newsImage" width="200" height="200">
```

"Lead Image Only" is in no way adequate. We need the ability to enter alt text for the lead image when uploaded, and that field needs to be required. See #259

**Comment by skleinfeldt** [comment]
Title can have the caption text (no markup).

Need a new field to define alt text and be required.

**Comment by alecpm** [comment]
Will require additional schemaextender and override for lead image viewlet. May also need to update `tiled view` which uses lead image explicitly (https://github.com/isawnyu/isaw.web/blob/f614553e524424bee0afe13a536f95d74900422e/src/isaw.theme/isaw/theme/browser/tiled_view.py)

**Comment by paregorios** [comment]
@cguardia this looks good, but I think it needs a couple of tweaks as follows:

 1. As @skleinfeldt noted above, please put caption text in the "title" attribute. Right now the "alt" text is being replicated in the "title" attribute, and WCAG recommends against that. If no caption text is provided by the user, then the "title" attribute should be suppressed entirely.

 2. Is it possible to make the "alt" text field required if a lead image is uploaded?

**Comment by cguardia** [comment]
Making it required took me a bit more time than I hoped, but it's done now. Please try it out.

**Comment by paregorios** [comment]
Moved into "reviewing" in the iteration project per comment from @cguardia [above](https://github.com/isawnyu/isaw.web/issues/296#issuecomment-413793011)

**Comment by paregorios** [comment]
This looks great.

**Comment by paregorios** [comment]
This is deployed to production and working, but one requirement has not been met:

[As requested above](https://github.com/isawnyu/isaw.web/issues/296#issuecomment-413290793), the "title" attribute should be suppressed entirely if no caption text has been entered.  

<img width="484" alt="screen shot 2018-08-21 at 9 41 53 am" src="https://user-images.githubusercontent.com/263285/44408715-898f2480-a526-11e8-8dfd-2842d16ea5e7.png">

[On my test page](https://isaw.nyu.edu/members/te20-40nyu.edu/lead-image-testing/test) I'm seeing an empty "title" attribute:

```html
<img src="https://isaw.nyu.edu/members/te20-40nyu.edu/lead-image-testing/test/leadImage_mini" alt="Cat yawning." title="" class="newsImage" width="120" height="120">
```

I would like to see:

```html
<img src="https://isaw.nyu.edu/members/te20-40nyu.edu/lead-image-testing/test/leadImage_mini" alt="Cat yawning." class="newsImage" width="120" height="120">
```

**Comment by cguardia** [comment]
At the moment, title is empty because we are sending an empty value to the archetypes tag method. If we simply omit the title, archetypes will use the object title by default always. Our options are:

1. Leave empty title attribute (least desirable, as I understand).
2. Do not send an empty string and let archetypes use the object's title.
3. Monkey patch archetypes tag method to not include a title if it comes empty.

**Comment by paregorios** [comment]
I will review WCAG guidance closely to see what makes the most sense.

**Comment by paregorios** [comment]
Documents I'm reading to figure out what we should do here:

 - [HTML 5.1 2nd Edition: 4.7.5.1 Embedded content: The img element: Requirements for providing text to act as an alternative for images](https://www.w3.org/TR/html51/semantics-embedded-content.html#alt-text)
 - [W3C Resources on Alternative Text for Images](https://www.w3.org/WAI/alt/#resources)

**Comment by paregorios** [comment]
@cguardia I think we need your behavior #3: monkey-patching the archetypes tag method

**Comment by paregorios** [comment]
Another tweak here, as per #259, never use title attribute on img tags, whether they are put inline through the rich text editor or via lead image.

**Comment by cguardia** [comment]
This is on staging now.

**Comment by paregorios** [comment]
This looks good. Please deploy.

**Comment by paregorios** [comment]
Hi @cguardia did this get deployed?

**Comment by cguardia** [comment]
Yes, I think it was.

**Comment by paregorios** [comment]
I am seeing the desired behavior when adding a new lead image on production, so we should consider this deployed. Closing.
